File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        data = []
        self.preorder(root, data, 0)
        return data

    def preorder(self, root: Optional[TreeNode], data: List[List[int]], height: int):
        if not root:
            return
        if len(data) == height:
            data.append([])

        data[height].append(root.val)

        self.preorder(root.left, data, height + 1)
        self.preorder(root.right, data, height + 1)   

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
    data := make([][]int, 0, 10)

	preorderSolve(root, &data, 0)

	return data
}

func preorderSolve(root *TreeNode, data *[][]int, height int) {
	if root == nil {
		return
	}
	if len(*data) == height {
		*data = append(*data, []int{})
	}
	(*data)[height] = append((*data)[height], root.Val)

	preorderSolve(root.Left, data, height+1)
	preorderSolve(root.Right, data, height+1)
}